src/scopeck.rs

```diff
@@ -1055,6 +1064,69 @@ pub(crate) fn scope_check(result: &mut ScopeResult, segments: &SegmentSet, names
     }
 }
 
+/// A [`Frame`] reference in the context of a [`Database`].
+/// This allows the values in the [`Frame`] to be resolved,
+#[derive(Copy, Clone, Debug)]
+pub struct FrameRef<'a> {
+    db: &'a Database,
+    frame: &'a Frame,
+}
+
+impl std::ops::Deref for FrameRef<'_> {
+    type Target = Frame;
+
+    fn deref(&self) -> &Self::Target {
+        self.frame
+    }
+}
+
+impl<'a> FrameRef<'_> {
+    /// Iterates over the essential hypotheses for this frame.
+    pub fn essentials(&'a self) -> impl Iterator<Item = (Label, &'a Formula)> + '_ {
```

Comment:
```suggestion
    pub fn essentials(&self) -> impl Iterator<Item = (Label, &'a Formula)> + '_ {
```
Alternatively, it could take the frame ref by value, since it is copy.